src/Http/Http.Abstractions/src/HttpProtocols.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Contains methods to verify the request protocol of an HTTP request. 
+    /// </summary>
+    public static class HttpProtocols
+    {
+        // We are intentionally using 'static readonly' here instead of 'const'.
+        // 'const' values would be embedded into each assembly that used them
+        // and each consuming assembly would have a different 'string' instance.
+        // Using .'static readonly' means that all consumers get these exact same
+        // 'string' instance, which means the 'ReferenceEquals' checks below work
+        // and allow us to optimize comparisons when these constants are used.
+
+        // Please do NOT change these to 'const'
+        public static readonly string Http10 = "HTTP/1.0";
+        public static readonly string Http11 = "HTTP/1.1";
+        public static readonly string Http2 = "HTTP/2";
+        public static readonly string Http3 = "HTTP/3";
```

Comment:
It seems a little unfortunate that we used `HTTP/2` for HTTP/2. The protocol string is `h2` (and `h3` for `HTTP/3`), though the friendly display name is indeed HTTP/3. The strings `HTTP/2` and `HTTP/3` never appear in the protocol (based on my reading).

But `HTTP/2` shipped already, so 🤷‍♂.

Reply:
HTTP/2 was the right choice. h2 is the ALPN protocol string, and that's not what we're doing with these values. You wouldn't want both h2 and h2c showing up here. HTTP/2 is the more general descriptor.